src/java/com/puppetlabs/puppetserver/ExecutionStubImpl.java

```diff
@@ -10,25 +14,78 @@
     private static final Logger log = LoggerFactory.getLogger(ExecutionStubImpl.class);
 
     /**
+     * Sets up an executor with some sane defaults that both executeCommand
+     * implementations depend on.
+     *
+     * @param errStream stream to write any errors during command execution
+     * @param outStream stream to write any output during command execution
+     * @return Executor with exit-code checking disabled and output and error streams setup
+     */
+    private static Executor setupExecutor(ByteArrayOutputStream errStream, ByteArrayOutputStream outStream) {
+        DefaultExecutor executor = new DefaultExecutor();
+        PumpStreamHandler streamHandler = new PumpStreamHandler(outStream, errStream);
+
+        // Don't throw exception on non-zero exit code
+        executor.setExitValues(null);
+
+        // Set up the handlers
+        executor.setStreamHandler(streamHandler);
```

Comment:
I think this bit with the `PumpStreamHandler` is the crucial bit of this implementation.  I took a quick glance at the code/docs and it seems to do what we want (i.e. what we were previously doing with `StreamBoozer`.  @cprice404 whatcha think?